nautobot_device_lifecycle_mgmt/management/commands/generate_app_test_data.py

```diff
@@ -0,0 +1,137 @@
+"""Generate test data for the Golden Config app."""
+
+import random
+
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+from django.db import DEFAULT_DB_ALIAS
+from nautobot.core.factory import get_random_instances
+from nautobot.dcim.models import Device, DeviceType, InventoryItem, Platform
+
+from nautobot_device_lifecycle_mgmt.choices import CVESeverityChoices
+from nautobot_device_lifecycle_mgmt.models import (
+    CVELCM,
+    ContactLCM,
+    ContractLCM,
+    DeviceSoftwareValidationResult,
+    HardwareLCM,
+    InventoryItemSoftwareValidationResult,
+    ProviderLCM,
+    SoftwareImageLCM,
+    SoftwareLCM,
+    ValidatedSoftwareLCM,
+    VulnerabilityLCM,
+)
+
+
+class Command(BaseCommand):
+    """Populate the database with various data as a baseline for testing (automated or manual)."""
+
+    help = __doc__
+
+    def add_arguments(self, parser):  # noqa: D102
+        parser.add_argument(
+            "--flush",
+            action="store_true",
+            help="Flush any existing data from the database before generating new data.",
+        )
+        parser.add_argument(
+            "--database",
+            default=DEFAULT_DB_ALIAS,
+            help='The database to generate the test data in. Defaults to the "default" database.',
+        )
+
```

Comment:
Should we add the optional `--seed` from the Nautobot generate_test_data so that it can be passed back to the command?

Reply:
I'm thinking that we'll just update the doc to tell the user to run the core `generate_test_data` first and then run this separately. Then if we rename this one to `generate_dlm_test_data` and rename golden config's to `generate_gc_test_data` they won't conflict with each other.

Reply:
That would be good, then if for some reason a user wants to run this and golden config in their test environment, they wouldn't override each other or flush each others data out.